File: WorldFeed/Server/src/WorldFeed.Common/Helpers/TaskDelayer.cs
namespace WorldFeed.Common.Helpers
{
    using System;
    using System.Threading.Tasks;

    public interface ITaskDelayer
    {
        Task Delay(TimeSpan timeSpan);
    }

    public class TaskDelayer : ITaskDelayer
    {
        public Task Delay(TimeSpan timeSpan)
        {
            return Task.Delay(timeSpan);
        }
    }
}

File: WorldFeed/Client/src/app/core/Core/Helpers/TaskDelayer.ts
import TimeSpan from "../../../c#-objects/TypeScript.NET-Core/packages/Core/source/Time/TimeSpan";

export interface ITaskDelayer {
  delay(timeSpan: TimeSpan): Promise<void>;
}

export class TaskDelayer implements ITaskDelayer {
  public delay(timeSpan: TimeSpan): Promise<void> {
    return Promise.Delay(timeSpan);
  }
}
